lms/djangoapps/instructor_task/tests/test_tasks_helper.py

```diff
@@ -202,3 +203,53 @@ def test_report_stores_results(self, mock_time):
 
                     self.assertEqual(return_val, UPDATE_STATUS_SUCCEEDED)
                     mock_store_rows.assert_called_once_with(self.course.id, filename, [test_header] + test_rows)
+
+
+class TestInstructorCourseForumsReport(TestReportMixin, InstructorTaskCourseTestCase):
+    """
+    Tests that course forums response report generation works.
+    """
+    def setUp(self):
+        self.course = CourseFactory.create(org=TEST_COURSE_ORG,
+                                           number=TEST_COURSE_NUMBER,
+                                           display_name=TEST_COURSE_NAME)
+
+        self.current_task = Mock()
+        self.current_task.update_state = Mock()
+
+    def tearDown(self):
+        if os.path.exists(settings.COURSE_FORUMS_USAGE_DOWNLOAD['ROOT_PATH']):
+            shutil.rmtree(settings.COURSE_FORUMS_USAGE_DOWNLOAD['ROOT_PATH'])
+
+    def test_report_fails_if_error(self):
+        with patch('instructor_task.tasks_helper.collect_course_forums_data') as mock_collect_data:
+            mock_collect_data.side_effect = KeyError
+
+            with patch('instructor_task.tasks_helper._get_current_task') as mock_current_task:
+                mock_current_task.return_value = self.current_task
+
+                self.assertEqual(push_course_forums_data_to_s3(None, None, self.course.id, None, 'generated'), UPDATE_STATUS_FAILED)
+
+    @patch('instructor_task.tasks_helper.datetime')
+    def test_report_stores_results(self, mock_time):
+        start_time = datetime.now(UTC)
+        mock_time.now.return_value = start_time
+
+        test_header = ['Date', 'Type', 'Number', 'Up Votes', 'Down Votes', 'Net Points']
+        test_rows = [['row1_field1', 'row1_field2'], ['row2_field1', 'row2_field2']]
+
+        with patch('instructor_task.tasks_helper._get_current_task') as mock_current_task:
+            mock_current_task.return_value = self.current_task
+
+            with patch('instructor_task.tasks_helper.collect_ora2_data') as mock_collect_data:
+                mock_collect_data.return_value = (test_header, test_rows)
+
+                with patch('instructor_task.models.LocalFSReportStore.store_rows') as mock_store_rows:
+                    return_val = push_course_forums_data_to_s3(None, None, self.course.id, None, 'generated')
+
+                    timestamp_str = start_time.strftime('%Y-%m-%d-%H%M')
+                    course_id_string = urllib.quote(self.course.id.to_deprecated_string().replace('/', '_'))
+                    filename = u'{}_course_forums_{}.csv'.format(course_id_string, timestamp_str)
+
+                    self.assertEqual(return_val, UPDATE_STATUS_SUCCEEDED)
```

Comment:
Just checking this I think is enough, the extra code for checking store_rows is kind of redundant in my opinion